pytypes/typechecker.pyi

```diff
@@ -0,0 +1,48 @@
+# Generated using mypy stubgen, then editing types for typechecked and
+# no_type_check.
+
+import __builtin__ as builtins
+import types
+import typing
+from . import type_util as type_util, util as util
+from .exceptions import InputTypeError as InputTypeError, OverrideError as OverrideError, ReturnTypeError as ReturnTypeError
+from .type_util import TypeAgent as TypeAgent, deep_type as deep_type, get_orig_class as get_orig_class, has_type_hints as has_type_hints, is_Tuple as is_Tuple, is_builtin_type as is_builtin_type, type_str as type_str
+from .util import getargspecs as getargspecs
+from typing import Any, Optional, Callable, TypeVar, Union, overload
+from warnings import warn as warn
+
+_python___import__ = builtins.__import__
+
+def _pytypes___import__(name: Any, globls: Optional[Any] = ..., locls: Optional[Any] = ..., fromlist: Any = ..., level: int = ...): ...
+
+class _DelayedCheck:
+    func: Any = ...
+    method: Any = ...
+    class_name: Any = ...
+    base_method: Any = ...
+    base_class: Any = ...
+    exc_info: Any = ...
+    raising_module_name: Any = ...
+    def __init__(self, func: Any, method: Any, class_name: Any, base_method: Any, base_class: Any, exc_info: Any) -> None: ...
+    def run_check(self, raise_NameError: bool = ...) -> None: ...
+
+def override(func: Any, auto: bool = ...): ...
+def typechecked_func(func: Any, force: bool = ..., argType: Optional[Any] = ..., resType: Optional[Any] = ..., prop_getter: bool = ...): ...
+def typechecked_class(cls, force: bool = ..., force_recursive: bool = ...): ...
+def typechecked_module(md: Any, force_recursive: bool = ...): ...
+
+T = TypeVar('T', Callable, property, type, types.ModuleType)
+@overload
+def typechecked(memb: str) -> Union[str, types.ModuleType]: ...
+@overload
+def typechecked(memb: T) -> T: ...
+def auto_override_class(cls, force: bool = ..., force_recursive: bool = ...): ...
+def auto_override_module(md: Any, force_recursive: bool = ...): ...
+def auto_override(memb: Any): ...
+def no_type_check(memb: T) -> T: ...
+def is_no_type_check(memb: Any): ...
+def check_argument_types(cllable: Optional[Any] = ..., call_args: Optional[Any] = ..., clss: Optional[Any] = ..., caller_level: int = ...): ...
+def check_return_type(value: Any, cllable: Optional[Any] = ..., clss: Optional[Any] = ..., caller_level: int = ...): ...
+
```

Comment:
Throughout this file, many type annotations are actually incorrect, i.e. (almost?) every ``Any``.
To name one example, ``auto_override_module(md: Any, `` should be something like ``auto_override_module(md: Union[str, types.ModuleType, ``. The original purpose of this was to annotate the decorators, i.e. ``typechecked``, ``override``, ``auto_override``. If you like, feel free to add more annotations but I think for every function there should either be a correct annotation or no annotation. So, please remove the auto-generated ones you did not work out to be actually correct.

Reply:
See my comment in https://github.com/Stewori/pytypes/pull/96#issuecomment-668280795 about updating the pyi to be more complete, and since then I have confirmed that a partial pyi with some definitions omitted causes "no such attribute" errors at least from mypy. The idea of including all those `Any` definitions is that it signals to the system that they do exist and we just don't know the types they use.

Moving it to an `__init__.pyi` helps a little with the bloat, but still I think having some `Any` definitions in there makes sense. Or I could spend more time trying to add useful types to everything. For type hints in .py you don't have that wrinkle of placeholder `Any` definitions, but they need to go in a separate .pyi if it needs to support python 3.4 or lower.

Reply:
Ahh, I expected this to be more flexible. It's better to keep this in a ``pyi`` file though. If one day a ``pyi`` is added for ``type_util`` a way is needed to prevent pytypes from self checking which would cause infinite recursion. ``@no_typecheck`` would be a way but that would also prevent static typechecking. pytypes could actively detect whether a to-be-checked function is from pytypes but that check slows down all typechecking and pytypes is already way too slow (because of such kind of checks etc). Having it in a ``pyi`` allows to have this detection in ``stubfile_manager``, i.e. it would simply not apply stubfiles from pytypes itself. This would keep the check out of the (non-initialization) runtime.
So, ``pyi`` is definitely the way to go.

Regarding the placeholder type annotations, given there's a compelling reason for them I am fine to keep them as is. However, please mark them as placeholders using some comment like
``# placeholder to prevent "no such attribute" errors in mypy; TODO: fill in proper types``
so we keep track what still needs to be done. Of course you are welcome to fill in properly as many types as you like, but it's no strict requirement for this PR then.

Reply:
K, I added a comment/TODO and a few more types. There aren't that many `Any` types remaining in the file actually, but I focused on the ones I'm familiar with and conservatively left some unspecified for this PR. If I guessed and picked a type that's too narrow, it could lead to annoying "wrong type" errors.